D/Dbus/build_tarballs.jl

```diff
@@ -33,18 +31,19 @@ make install
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
 platforms = filter!(p -> Sys.islinux(p) || Sys.isfreebsd(p), supported_platforms())
-# TODO: Remove this restriction for the next build
-filter!(p -> arch(p) != "armv6l", platforms)
 
+# The products that we will ensure are always built
 products = [
     LibraryProduct("libdbus-1", :libdbus),
     ExecutableProduct("dbus-daemon", :dbus_daemon)
 ]
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-    Dependency("Expat_jll"; compat="2.2.7"),
+    Dependency(PackageSpec(name="Expat_jll", uuid="2e619515-83b5-522b-bb60-26c02a35a201"); platforms=platforms),
```

Comment:
Why did you remove the compat entry from expat here?

Reply:
Also, what's the point of `platforms=platforms`?  You aren't really filtering anything, as far as I can tell.

Reply:
For the `platforms=platforms`, it was my ignorance on how it interacted with the option on `build_tarballs`. I'll delete it.

As for the first, the only mention I've found about the versions that `dbus` supports for `expat` is [this](https://cgit.freedesktop.org/dbus/dbus/tree/NEWS?h=master#n1603); in which they mention only that "Expat >= 2.1.0 is required". Given that `dbus` builds correctly against the latest version, is there a way to test that this build does not break some other package?

Reply:
The compat should still be restored, or adapted, but definitely not removed...

Reply:
Update it. Is it OK to set it to `"2.1.0, 3"`, or, do you have any specific suggestion?

Reply:
I think that should have done it. Do you think is it related to [`BinaryBuilder.jl#1330`](https://github.com/JuliaPackaging/BinaryBuilder.jl/issues/1330) or [`Yggdrasil#8211`](https://github.com/JuliaPackaging/Yggdrasil/issues/8211)?

Reply:
I'm totally missing what's the bug here and how anything is possibly related to the linked tickets.

Reply:
>  Is it OK to set it to "^2.1"; or, do you have any specific suggestion?

Why don't you just keep the existing one?

Reply:
> I'm totally missing what's the bug here and how anything is possibly related to the linked tickets.

According to [#`BinaryBuilderBase.Dependency`](https://docs.binarybuilder.org/stable/reference/#BinaryBuilderBase.Dependency)

> The optional keyword argument `compat` can be used to specify a string for use in the `Project.toml` of the generated Julia package.

Doesn't that means that setting `compat="^2.1"` should set the `[compat]` of `Expat_jll` in the `Project.toml` of `nativefiledialog_jll.jl` to `Expat_jll = "^2.1"`? Which, according to [#Caret-specifiers](https://pkgdocs.julialang.org/v1/compatibility/#Caret-specifiers) implies the versions of `Expat_jll` accepted are `[2.1.0, 3.0.0)`?

But, the `buildkite` log says that:

```julia
ERROR: LoadError: Unsatisfiable requirements detected for package Expat_jll [2e619515]:
 Expat_jll [2e619515] log:
 ├─possible versions are: 2.2.7-2.6.2 or uninstalled
 └─restricted to versions 2.1.0 by an explicit requirement — no versions left
```

I'll change it to the current one and left it like that; but, I don't understand where I'm interpreting wrong the docs. 😓

Reply:
Given that I don't see where I'm wrong in reading the docs, I searched issues for errors with `Dependency` and `compat` bounds. That is why I landed to those I shared, though I did not research further, which is my mistake.

Reply:
Try to read more:

> If compat is non-empty and build_version is not passed, the latter defaults to the minimum version compatible with the compat specifier.

Reply:
I changed it to the latest version of `Expat_jll`, as the old one did not have support for `armv6l` as seen in [`buildkite`](https://buildkite.com/julialang/yggdrasil/builds/11261). Now in green for all platforms.

Reply:
> Try to read more:
> 
> > If compat is non-empty and build_version is not passed, the latter defaults to the minimum version compatible with the compat specifier.

I'll be more careful, thanks for the answer.